python/lsst/meas/algorithms/detection.py

```diff
@@ -489,6 +515,38 @@ def setEdgeBits(maskedImage, goodBBox, edgeBitmask):
                                                      afwGeom.ExtentI(w, h)), afwImage.LOCAL)
             edgeMask |= edgeBitmask
 
+    @pipeBase.timeMethod
+    def replacePeaks(self, targetFpList, goodPeakFpList):
+        """Replace all of the peaks in one FootprintSet with the overlapping ones from another.
+
+        When this would remove all of the Peaks from a Footprint, we instead remove all but the
+        brightest.
+
+        Parameters
+        ----------
+        targetFpSet : `list`
+            Sequence containing the Footprints whose Peaks we want to replace.
+        goodPeakFpSet : `list`
+            Sequence containing the Footprints whose Peaks we want to use instead.
+        """
+        for targetFp in targetFpList:
+            targetBBox = targetFp.getBBox()
+            newPeaks = afwDet.PeakCatalog(targetFp.getPeaks().getTable())
+            for goodPeakFp in goodPeakFpList:
+                if not targetBBox.overlaps(goodPeakFp.getBBox()):
+                    continue
+                for candidatePeak in goodPeakFp.getPeaks():
+                    if targetFp.contains(candidatePeak.getI()):
+                        newPeaks.append(candidatePeak)
```

Comment:
It will break memory contiguity, but I believe in practice we'll copy over the vast majority of the new peaks in the same order they were allocated and remove the vast majority of the old peaks, so in the absence of profiling I don't think there's a strong case to be made on deep vs. shallow copy here.